ctru-rs/src/services/am.rs

```diff
@@ -0,0 +1,121 @@
+use crate::error::ResultCode;
+use crate::services::fs::FsMediaType;
+use std::marker::PhantomData;
+use std::mem::MaybeUninit;
+
+#[derive(Copy, Clone, Debug)]
+#[repr(C)]
+pub struct TitleInfo {
+    id: u64,
+    size: u64,
+    version: u16,
+    pad: u16,
+    type_: u32,
+}
+
+// Make sure TitleInfo is correct size
+const _TITLEINFO_SIZE_CHECK: [u8; 0x18] = [0; std::mem::size_of::<TitleInfo>()];
+
+impl TitleInfo {
+    pub fn id(&self) -> u64 {
+        self.id
+    }
+    pub fn size_bytes(&self) -> u64 {
+        self.size
+    }
+    pub fn version(&self) -> u16 {
+        self.version
+    }
+    pub fn type_(&self) -> u32 {
```

Comment:
This isn't referenced in `libctru`'s docs, could you please explain what it does (and maybe link some references)?

Reply:
I was sorta using Citra's sources as a reference to some structs and forgot to remove this once i noticed it's not really used anywhere else

Reply:
No problem. I suggest giving a look at [libctru.devkitpro.org](https://libctru.devkitpro.org/) for `libctru` references and at [3dbrew.org](https://3dbrew.org) for specific system knowledge (it's helpful to learn how the hardware works). Citra doesn't really seem like the best source, especially since it's an inaccurate emulator mainly used for games, and not for development.

Reply:
I usually look at `ctru-sys/src/bindings.rs` and 3dbrew.org for documentation, as I find doxygen confusing to use